src/include/parser/expression/abstract_expression.h

```diff
@@ -38,6 +52,21 @@ class AbstractExpression {
    */
   AbstractExpression() = default;
 
+  /**
+   * @param expression_name Set the expression name of the current expression
+   */
+  void SetExpressionName(std::string expression_name) { expression_name_ = std::move(expression_name); }
+
+  /**
+   * @param return_value_type Set the return value type of the current expression
+   */
+  void SetReturnValueType(type::TypeId return_value_type) { return_value_type_ = return_value_type; }
+
+  /**
+   * @param depth Set the depth of the current expression
+   */
+  void SetDepth(int depth) { depth_ = depth; }
```

Comment:
The binder will be using them. The reason they are protected instead of private is because subclasses of abstract_expression will be able to inherit and use them. For example, SetReturnValueType() is used in aggregate_expression and comparison_expression's DeduceReturnValueType()

Reply:
Ahh I missed the `protected` keyword. That sounds good.